internal/locate/region_request.go

```diff
@@ -573,18 +573,40 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 	if state.option.preferLeader {
 		state.lastIdx = state.leaderIdx
 	}
+	offset := rand.Intn(replicaSize)
+	reloadRegion := false
 	for i := 0; i < replicaSize && !state.option.leaderOnly; i++ {
-		idx := AccessIndex((int(state.lastIdx) + i) % replicaSize)
-		// If the given store is abnormal to be accessed under `ReplicaReadMixed` mode, we should choose other followers or leader
-		// as candidates to serve the Read request. Meanwhile, we should make the choice of next() meet Uniform Distribution.
-		for cnt := 0; cnt < replicaSize && !state.isCandidate(idx, selector.replicas[idx]); cnt++ {
-			idx = AccessIndex((int(idx) + rand.Intn(replicaSize)) % replicaSize)
+		var idx AccessIndex
+		if state.option.preferLeader {
+			if i == 0 {
+				// In prefer leader mode, the selector choose leader firstly, then randomly choose other followers.
+				idx = AccessIndex((int(state.lastIdx) + i) % replicaSize)
+			} else {
+				if state.option.preferLeader && (i+offset)%replicaSize == 0 {
+					// skip leader because prefer leader already tried it.
+					offset++
+				}
+				idx = AccessIndex((int(state.lastIdx) + i + offset) % replicaSize)
+			}
+		} else {
+			// If the given store is abnormal to be accessed under `ReplicaReadMixed` mode, we should choose other followers or leader
+			// as candidates to serve the Read request. Meanwhile, we should make the choice of next() meet Uniform Distribution.
+			idx = AccessIndex((int(state.lastIdx) + i + offset) % replicaSize)
```

Comment:
It's a bit more complicated. Recommend just directly use the previous commit.

And here, `PreferLeader` is a Read mode which select the leader replica as priority, and only if is leader slow or abnormal, the followers will be chosen.

```suggestion
		// If the given store is abnormal to be accessed under `ReplicaReadMixed` mode, we should choose other followers or leader
		// as candidates to serve the Read request. Meanwhile, we should make the choice of next() meet Uniform Distribution.
		idx := AccessIndex((int(state.lastIdx) + i + offset) % replicaSize)
```